celements-xwiki-core/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -1830,7 +1837,9 @@ public String getLanguagePreference(XWikiContext context) {
           return language;
         }
       }
-    } catch (XWikiException e) {}
+    } catch (Exception exp) {
```

Comment:
```suggestion
    } catch (XWikiException exp) {
```
keep the catch as XWikiException